hide only rule dialogue turns (#7701)
RasaHQ/rasa: docs/docs/rules.mdx

This example rule applies at the start of conversation as well as when the user decides
to a send a message with an intent `greet` in the middle of an ongoing conversation.

Dialogue turns that only appear as rules in the training data and do not appear in stories
will be ignored by ML-only policies like `TEDPolicy` at prediction time.

> For example if you define the greeting rule as above and don't add it to any of your stories,

Reviewer: Let's give actual rules / stories as example to make it less abstract for users.

This example rule applies at the start of conversation as well as when the user decides
to a send a message with an intent `greet` in the middle of an ongoing conversation.

> Dialogue turns that only appear as a rule in the training data and do not appear in stories
> will be hidden in the tracker states for ML only policies like `TEDPolicy`.

Reviewer 2: ```suggestion
Dialogue turns that only appear as rules in the training data and do not appear in stories
will be ignored by ML-only policies like `TEDPolicy` at prediction time.
```

Reviewer 2: Is this still a valid explanation?

> For example if you have a greet rule above and don't add it to any of your stories then
> when during prediction `RulePolicy` predicts `utter_greet`, afterwards `TEDPolicy` will
> see dialogue turns as if user never input intent `greet`.
> 

Reviewer 2: ```suggestion
For example if you define the greeting rule as above and don't add it to any of your stories,
after `RulePolicy` predicts `utter_greet`, `TEDPolicy` will
make predictions as if the `greet, utter_greet` turn did not occur.
```
